tests/accessor/local_accessor_constructors.h

```diff
@@ -0,0 +1,114 @@
+/*******************************************************************************
+//
+//  SYCL 2020 Conformance Test Suite
+//
+//  Provides tests for sycl::local_accessor constructors
+//
+*******************************************************************************/
+#ifndef SYCL_CTS_LOCAL_ACCESSOR_H
+#define SYCL_CTS_LOCAL_ACCESSOR_H
+#include "catch2/catch_test_macros.hpp"
+
+#include "accessor_common.h"
+
+namespace local_accessor_constructors {
+using namespace sycl_cts;
+using namespace accessor_tests_common;
+
+constexpr accessor_type AccType = accessor_type::local_accessor;
+
+template <typename DataT, int Dimension>
+void test_default_constructor(const std::string& type_name) {
+  const auto section_name =
+      get_section_name<Dimension>(type_name, "Default constructor");
+
+  SECTION(section_name) {
+    auto get_acc_functor = [] {
+      return sycl::local_accessor<DataT, Dimension>();
+    };
+    if constexpr (std::is_const_v<DataT>) {
+      check_def_constructor<AccType, DataT, Dimension, sycl::access_mode::read,
+                            sycl::target::device>(get_acc_functor);
+    } else {
+      check_def_constructor<AccType, DataT, Dimension,
+                            sycl::access_mode::read_write,
+                            sycl::target::device>(get_acc_functor);
+    }
+  }
+}
+
+template <typename DataT>
+void test_zero_dimension_buffer_constructor(const std::string& type_name) {
+  const auto section_name =
+      get_section_name<0>(type_name, "Zero dimension constructor");
+
+  SECTION(section_name) {
+    auto get_acc_functor = [](sycl::buffer<DataT, 1>& data_buf,
+                              sycl::handler& cgh) {
+      return sycl::local_accessor<DataT, 0>(cgh);
+    };
+    if constexpr (std::is_const_v<DataT>) {
+      check_zero_dim_constructor<AccType, DataT, sycl::access_mode::read,
+                                 sycl::target::device>(get_acc_functor);
+    } else {
+      check_zero_dim_constructor<AccType, DataT, sycl::access_mode::read_write,
+                                 sycl::target::device>(get_acc_functor);
+    }
+  }
+}
+
+template <typename DataT, int Dimension>
+void test_common_constructors(const std::string& type_name) {
+  auto r = util::get_cts_object::range<Dimension>::get(1, 1, 1);
+  auto offset = sycl::id<Dimension>();
+
+  auto section_name =
+      get_section_name<Dimension>(type_name, "From sycl::range constructor");
+
+  SECTION(section_name) {
+    auto get_acc_functor = [r](sycl::buffer<DataT, Dimension>& data_buf,
+                               sycl::handler& cgh) {
+      return sycl::local_accessor<DataT, Dimension>(r, cgh);
+    };
+    if constexpr (std::is_const_v<DataT>) {
+      check_common_constructor<AccType, DataT, Dimension,
+                               sycl::access_mode::read, sycl::target::device>(
+          get_acc_functor);
```

Comment:
I'm a bit curious why don't we use `AccType` within `check_common_constructor` to differentiate between different use cases instead of `Target`. Or why don't we have partial specializations with `check_common_constructor` being a struct instead of function (to avoid issues with parameter overloading).
Still looks like we have a bunch of tests in open PRs already, so I suppose any updates might result in too many changes within interdependent PRs. Any thoughts?

Reply:
These functions can be refactored, but it is too difficult to do, when there are at least 5 PRs, that all depend on one "common" file. IMO it will be easier to do when all PRs that depend on #318 will be merged, and we can provide refactoring with separate PR.

Reply:
Yes.